### ISSUE TITLE
influxdb network is unreachable (local:docker runner)

### ISSUE THREAD

**Issue by nonsense** [testground/testground#1006]
If `sidecar` container is running and `influxdb` container is not, the healthcheck would start influxdb, but the routes would not work, because currently sidecar resolves dependencies on startup and not on every run.

```
May 13 11:46:35.070731  INFO    32.6513s    MESSAGE << c5728c35817f >> failed to send batch to InfluxDB; attempt 4; err: Post http://testground-influxdb:8086/write?consistency=&db=testground&precision=ns&rp=: dial tcp 192.18.0.5:8086: connect: network is unreachable      {"req_id": "d67a7590"}
```

**Comment by raulk** [comment]
We need to strike a balance between (1) resolving each service every time we manage a container (in a 10k run that means 20k or 30k DNS requests in a very short amount of time, something that k8s and docker nameservers don't appreciate), (2) and resolving only once at the start of the sidecar.

Potential fix could be to cache resolved IPs, and for every container we instrument, ping the IPs to validate them, re-resolving on failure.

Could probably use https://github.com/rs/dnscache and/or the singleflight package: https://pkg.go.dev/golang.org/x/sync/singleflight?tab=doc to help deduplicate a failure storm.

**Comment by nonsense** [comment]
> (1) resolving each service every time we manage a container (in a 10k run that means 20k or 30k DNS requests in a very short amount of time, something that k8s and docker nameservers don't appreciate), (2) and resolving only once at the start of the sidecar.

Actually I think (1) is not going to work, because if a required/optional service `moves` in the middle of a run we have a problem that is hard to address - we have already resolved the previous IP, and we have blacklisted all other IPs, but the service IP.

Hence if a service (let's say InfluxDB) moves in the middle of us allowing/blocking network paths for a given test run, we have to start from scratch for all containers in the test run, which makes the whole thing rather complicated.

---

I think for the short/mid-term an OK solution would be to just resolve all services once at the beginning of any test run - this will work fine even if we have concurrent test runs, which we are not planning right now.
